Updated documentation for UI related to Material Design implementation

rmathis opened the pull request:
Updated documentation for user interface related to Material Design implementation (#4730). Also included update to supported browsers (swapped IE for Edge) and added section regarding Thymeleaf and documentation. Added links to UI libraries for reference purposes.

hdeadman commented:
I noticed IE 11 didn't look good (in my customized 6.2 overlay) but hadn't gotten around to investigating. Is there no hope for supporting IE11?

rmathis commented:
> I noticed IE 11 didn't look good (in my customized 6.2 overlay) but hadn't gotten around to investigating. Is there no hope for supporting IE11?

Given the libraries (bootstrap and material.io) used, it is possible to better support IE11, but there has not been any effort to improve the user experience in that browser.

Perhaps I should update the wording to say what is "explicitly supported", and say that other browsers may be supported with additional effort in the future? Or remove the support section update from this pull request until we have time to work on IE11 support?

rmathis commented:
I've done a bit of work to make sure the layout and color scheme work with IE11, and fixed the javascript errors popping up in that browser. I've updated the browser support list to reflect this as well. Ref #4880

mmoayyed commented on docs/cas-server-documentation/ux/User-Interface-Customization-Views.md:
```diff
@@ -30,6 +29,12 @@ Upon a successful `200` status result, the response body is expected to contain
 
+## Thymeleaf
+
+CAS uses [Thymeleaf (v3.x)](https://www.thymeleaf.org) for its markup rendering engine. Each template is decorated by `layout.html` template file, which provides a layout structure for the template's content. Individual components optimized for re-use among multiple templates are stored in the `src/main/resources/templates/fragments` folder, and referenced by the templates in `src/main/resources/ ...
```
Avoid using version numbers. There is no guarantee that link points to v3, or that it will 2 weeks from now.

mmoayyed commented on docs/cas-server-documentation/ux/User-Interface-Customization-Views.md:
```diff
@@ -30,6 +29,12 @@ Upon a successful `200` status result, the response body is expected to contain
+
+CAS uses [Thymeleaf (v3.x)](https://www.thymeleaf.org) for its markup rendering engine. Each template is decorated by `layout.html` template file, which provides a layout structure for the template's content. Individual components optimized for re-use among multiple templates are stored in the `src/main/resources/templates/fragments` folder, and referenced by the templates in `src/main/resources/ ...
+
+Refer to the [Thymeleaf documentation](https://www.thymeleaf.org/doc/tutorials/3.0/usingthymeleaf.html) for more information on its use and syntax.
```
```suggestion
Refer to the [Thymeleaf documentation](https://www.thymeleaf.org) for more information on its use and syntax.
```
